src/components/AccordionDescription.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useState } from "react";
+import { TermBase, useTerm } from "../api/TermAPI";
+import { emptyTerm } from "./TermPage";
+import { Box, CircularProgress, Typography } from "@material-ui/core";
+
+const AccordionDescription: React.FC<TermBase> = (props) => {
+  const [description, setDescription] = useState<string>();
+  const { data = [], isLoading, isSuccess } = useTerm(props ?? emptyTerm);
+
+  useEffect(() => {
```

Comment:
I don't think you need this effect at all. What you can do instead is just put this right before the `if (isSuccess)` render:

```
const description = data.definition?.cs ? data.definition.cs : "Pojem nemá definici"
```

It will be even more efficient, because setting the description as the component state triggers another render, which is unnecessary.

The rule of thumb is that if you can calculate something from a data that you already have, then you just do that. If you need some heavy calculations, then there is the `useMemo` hook. But having an internal state is almost always unnecessary.